pandas/tests/copy_view/test_methods.py

```diff
@@ -214,3 +214,20 @@ def test_chained_methods(request, method, idx, using_copy_on_write):
     df.iloc[0, 0] = 0
     if not df2_is_view:
         tm.assert_frame_equal(df2.iloc[:, idx:], df_orig)
+
+
+def test_set_index(using_copy_on_write):
+    # GH 49473
+    df = DataFrame(
+        {
+            "month": [1, 4, 7, 10],
```

Comment:
While there is nothing wrong with this test dataframe, but just to keep it simple, I would suggest to use the same dataframe as much as possible in all tests in this file, and so use `df = DataFrame({"a": [1, 2, 3], "b": [4, 5, 6], "c": [0.1, 0.2, 0.3]})` that I have been using above.